### ISSUE TITLE
[BUG] in GET /v2/entities with "q" query parameter returns wrong results

### ISSUE THREAD

**Issue by iariasleon** [telefonicaid/fiware-orion#1592]
#### Create Requests
```
POST http://localhost:1026/v2/entities
Fiware-Service: qp_with_binary
Fiware-ServicePath: /test
{"id": "room_6", "my_value": 75}
```

```
POST http://localhost:1026/v2/entities
Fiware-Service: qp_with_binary
Fiware-ServicePath: /test
{"id": "room_5", "my_value": 45}
```

#### Mongo doc:
```
{ "_id" : { "id" : "room_5", "servicePath" : "/test" }, "attrNames" : [ "my_value" ], "attrs" : { "my_value" : { "type" : "", "creDate" : 1449084032, "modDate" : 1449084032, "value" : 45 } }, "creDate" : 1449084032, "modDate" : 1449084032 }
{ "_id" : { "id" : "room_6", "servicePath" : "/test" }, "attrNames" : [ "my_value" ], "attrs" : { "my_value" : { "type" : "", "creDate" : 1449084042, "modDate" : 1449084042, "value" : 75 } }, "creDate" : 1449084042, "modDate" : 1449084042 }
```

#### Get Request
```
GEThttp://localhost:1026/v2/entities?q=my_value>=
Fiware-Service: qp_with_type
Fiware-ServicePath: /test
```

#### Dataset
```
q=my_value>=
q=my_value>
q=my_value!=
```

#### Get Response
```
Status: 200 OK
[{"id":"room_5","my_value":45.000000},{"id":"room_6","my_value":75.000000}]
```

**Comment by kzangeli** [comment]
It might be that the parser of 'q' has a stupid bug doing that >= is interpreted as BIGGER THAN '=', instead of 'BIGGER OR EQUAL TO' ...

**Comment by iariasleon** [comment]
I think that if a binary expression did not have the three elements (attribute, operator and value) in that order, the parser should return an error, the same for the unary expressions, which should have two elements (operator and type or attribute) and determine if the unary operator is allowed only in front or behind or in front no matter, if so it should be in the documentation.

**Comment by raconte** [comment]
1.-Create entity 
(curl localhost:1026/v2/entities -s -S --header Content-Type:application/json --header 'Accept: application/json' -d @- | python -mjson.tool)<< EOF
{
  "type": "Apartment",
  "id": "Floor3",
  "temperature": {
    "value": 24.4
  }
}  
EOF
 
(curl localhost:1026/v2/entities -s -S --header Content-Type:application/json --header 'Accept: application/json' -d @- | python -mjson.tool)<< EOF
{
  "type": "Apartment",
  "id": "Floor4",
  "temperature": {
    "value": 27.4
  }
}  
EOF

3.- GET 
(curl localhost:1026/v2/entities?q=value>= -s -S --header Content-Type:application/json --header 'Accept: application/json'| python -mjson.tool)

Nothing is returned

4.-If you specified the type 
(curl localhost:1026/v2/entities?&temperature>=24.5 -s -S --header Content-Type:application/json --header 'Accept: application/json'| python -mjson.tool)

Is returned both.

**Comment by fgalan** [comment]
Fixed in PR https://github.com/telefonicaid/fiware-orion/pull/1735. Please @iariasleon have a look in order to check it is fixed.

**Comment by fgalan** [comment]
Issue fixed in PR #1738. Please @iariasleon have check if it has been finaly solved.

**Comment by iariasleon** [comment]
LGTM 
```
 http code: 400
 content-length: 63
 {"error":"BadRequest","description":"invalid query expression"}
```
